openmp/runtime/src/kmp_affinity.h

```diff
@@ -1098,6 +1098,50 @@ class kmp_hw_subset_t {
     qsort(items, depth, sizeof(item_t), hw_subset_compare);
   }
   bool specified(kmp_hw_t type) const { return ((set & (1ull << type)) > 0); }
+
+  // Canonicalize the KMP_HW_SUBSET value if it is not an absolute subset.
+  // This means putting each of {sockets, cores, threads} in the topology if
+  // they are not specified:
+  // e.g., 1s,2c => 1s,2c,*t | 2c,1t => *s,2c,1t | 1t => 1s,*c,*t | etc.
```

Comment:
Is `1t => 1s,*c,*t` the right canonicalization?

Reply:
Thanks! fixed.